applications/audio/broadcast_config_tool/src/main.c

```diff
@@ -1279,6 +1467,93 @@ static int cmd_program_info(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+#define FILE_LIST_BUF_SIZE 512
+static int cmd_file_list(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	char buf[FILE_LIST_BUF_SIZE];
+	size_t buf_size = FILE_LIST_BUF_SIZE;
+	char *dir_path = NULL;
+
+	if (argc > 2) {
+		shell_error(shell, "Usage: bct file list [dir path]");
+		return -EINVAL;
+	}
+
+	if (argc == 2) {
+		dir_path = argv[1];
+	}
+
+	ret = sd_card_list_files(dir_path, buf, &buf_size);
+	if (ret) {
+		shell_error(shell, "List files err: %d", ret);
+		return ret;
+	}
+
+	shell_print(shell, "%s", buf);
+
+	return 0;
+}
+
+static int cmd_file_select(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_idx;
+	uint8_t sub_idx;
+	uint8_t bis_idx;
+
+	if (argc < 4) {
```

Comment:
Same here I suppose, `if (argc != 4) {`

Reply:
I see now that is not that straight forward.

Reply:
Hmm isn't it? I think you're right on this one - anything else than 4 arguments is a misuse. (we could just ignore the extra arguments, but I think it's better to be strict on this)

Reply:
Now it stands as !=5. Is this correct?